python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -1525,14 +1525,52 @@ def mexists(
         n_chunks = 0
         for chunk in chunk_iterable(refs, chunk_size=chunk_size):
             chunk_result = self._mexists(chunk, artifact_existence)
-            if log.isEnabledFor(VERBOSE):
-                n_results = len(chunk_result)
-                n_checked += n_results
+
+            # The log message level and content depend on how many
+            # datasets we are processing.
+            n_results = len(chunk_result)
+
+            log_threshold = VERBOSE
+            n_checked += n_results
+
+            # This sum can take some time so only do it if we know the
+            # result is going to be used.
+            n_found = 0
+            if log.isEnabledFor(log_threshold):
                 # Can treat the booleans as 0, 1 integers and sum them.
                 n_found = sum(chunk_result.values())
                 n_found_total += n_found
-                log.verbose(
-                    "Number of datasets found in datastore for chunk %d = %d/%d (running total: %d/%d)",
+
+            # We are deliberately not trying to count the number of refs
+            # provided in case it's in the millions. This means there is a
+            # situation where the number of refs exactly matches the chunk
+            # size and we will switch to the multi-chunk path even though
+            # we only have a single chunk.
+            if n_results < chunk_size and n_chunks == 0:
+                # Single chunk will be processed so we can provide more detail.
+                if n_results == 1:
+                    ref = list(chunk_result)[0]
+                    log.log(
+                        log_threshold,
+                        "Calling mexists() with single ref that does%s exist (%s).",
+                        "" if chunk_result[ref] else " not",
+                        ref,
+                    )
+                else:
+                    # Single chunk but multiple files. Summarize.
+                    log.log(
+                        log_threshold,
```

Comment:
Maybe use VERBOSE here if the chunk took more than 10 seconds, else DEBUG? cc/ @TallJimbo

Reply:
Changing logging level depending on how long the code runs may be confusing, people may be surprised if messages start appearing/disappearing randomly.

Reply:
So leave it so that you get a debug log message if you have one dataset but verbose log message if you have more than one.

Reply:
I should add that we use `mexists()` in `SingleQuantumExecutor` and also turn on VERBOSE logging, so we are going to get this log message from there every time a quantum has more than one input (and most of the time there won't be 10,000 inputs). The 10,000 path comes up with `butler transfer-datasets` at the end of the workflow.

Reply:
OK, if VERBOSE is the default for many cases then it should be OK to avoid logging for single ref case.